src/util/tmpl/fd_treap.c

```diff
@@ -625,7 +628,7 @@ TREAP_(idx_query)( TREAP_(t) const * treap,
   return i;
```

Comment:
TREAP_QUERY_T might be different from TREAP_T. Is the different cmp function meant to deal with this?

Reply:
It is possible (see the types of queries done in the treaps used for workspace) to do queries over a range instead of an exact match.  So, an element of treap doesn't specify one possible query.  It can specify a disjoint range of queries satisfied by that element.  This in turn implies the need for some flexibility on the API presented to the user for queries versus how queries are done for internal operations.